data/api/application-service/definitions/namespace_list.yaml

```diff
@@ -19,10 +19,13 @@ items:
   properties:
     regex:
       type: string
-      description: A POSIX regular expression defining which values this namespace includes.
+      description: >-
```

Comment:
> removes spurious line breaks in the descriptions by switching from literal strip (|-) to folded strip (>-) YAML string representation (https://yaml-multiline.info/)

The descriptions are written using Markdown (just like most of the spec) so we don't really care about "spurious" line breaks, they should be removed by the Markdown parser.

In my opinion, switching to the folded strip is worse, because the text does not behave as Markdown. For example we need to put 2 empty lines between paragraphs.

Reply:
https://github.com/element-hq/synapse/pull/17892#discussion_r2086721314 is relevant to this discussion. I expect whatever solution we end up using there, can be useful here.

Reply:
I'm in agreement with @richvdh.

Reply:
-0.5 from me as well - on top of having to remember about triple linebreaks, they blow up the definition file vertically. I don't find it worth the effort trying to write Markdown so that both Markdown and plain-text processing tools could render it nicely. Simply try to add a bullet list to the example above and see how YAML processing smashes line breaks that are now-valuable for Markdown; and it's still not the last gotcha on this path.